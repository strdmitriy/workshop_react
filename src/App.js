import React from 'react';
import ListTabs from './components/ListTabs';

const App = () => (
  <ListTabs />
);

export default App;
