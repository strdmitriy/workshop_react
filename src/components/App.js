import React from 'react';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import "react-tabs/style/react-tabs.css";

export default () => (
  <Tabs>
    <TabList>
      <Tab data="tab-1">Title 1</Tab>
      <Tab data="tab-2">Title 2</Tab>
    </TabList>

    <TabPanel>
      <h2>Any content 1</h2>
    </TabPanel>
    <TabPanel>
      <h2>Any content 3</h2>
    </TabPanel>
  </Tabs>
);
