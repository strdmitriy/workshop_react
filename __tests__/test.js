import React from 'react';
import App from '../src/App';

describe('Tabs', () => {
  const dataTest = 'li[data-test^="tab"]';
  it('renders correctly', () => {
    const content = mount(<App />);
    expect(content).toMatchSnapshot();
  });

  it('active tab-1', () => {
    const content = mount(<App />);
    const tab = content.find(dataTest).at(0);
    console.log(tab);
    tab.simulate('click');
    expect(content.render()).toMatchSnapshot();
  });

  it('active tab-2', () => {
    const content = mount(<App />);
    const tab = content.find(dataTest).at(1);
    tab.simulate('click');
    expect(content.render()).toMatchSnapshot();
  });
});
