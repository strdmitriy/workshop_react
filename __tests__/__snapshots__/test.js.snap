// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs active tab-1 1`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      data-test="tab-1"
      id="react-tabs-4"
      role="tab"
      tabindex="0"
    >
      Title 1
    </li>
    <li
      aria-controls="react-tabs-7"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-test="tab-2"
      id="react-tabs-6"
      role="tab"
    >
      Title 2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-5"
    role="tabpanel"
  >
    <h2>
      Any content 1
    </h2>
  </div>
  <div
    aria-labelledby="react-tabs-6"
    class="react-tabs__tab-panel"
    id="react-tabs-7"
    role="tabpanel"
  />
</div>
`;

exports[`Tabs active tab-2 1`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-9"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-test="tab-1"
      id="react-tabs-8"
      role="tab"
    >
      Title 1
    </li>
    <li
      aria-controls="react-tabs-11"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      data-test="tab-2"
      id="react-tabs-10"
      role="tab"
      tabindex="0"
    >
      Title 2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-8"
    class="react-tabs__tab-panel"
    id="react-tabs-9"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-10"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-11"
    role="tabpanel"
  >
    <h2>
      Any content 3
    </h2>
  </div>
</div>
`;

exports[`Tabs renders correctly 1`] = `
<App>
  <ListTabs>
    <Tabs
      defaultFocus={false}
      defaultIndex={null}
      forceRenderTabPanel={false}
      selectedIndex={null}
    >
      <UncontrolledTabs
        className="react-tabs"
        focus={false}
        forceRenderTabPanel={false}
        onSelect={[Function]}
        selectedIndex={0}
      >
        <div
          className="react-tabs"
          data-tabs={true}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <TabList
            className="react-tabs__tab-list"
            key=".0"
          >
            <ul
              className="react-tabs__tab-list"
              role="tablist"
            >
              <Tab
                className="react-tabs__tab"
                data-test="tab-1"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-0"
                key=".0"
                panelId="react-tabs-1"
                selected={true}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-1"
                  aria-disabled="false"
                  aria-selected="true"
                  className="react-tabs__tab react-tabs__tab--selected"
                  data-test="tab-1"
                  id="react-tabs-0"
                  role="tab"
                  tabIndex="0"
                >
                  Title 1
                </li>
              </Tab>
              <Tab
                className="react-tabs__tab"
                data-test="tab-2"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-2"
                key=".1"
                panelId="react-tabs-3"
                selected={false}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-3"
                  aria-disabled="false"
                  aria-selected="false"
                  className="react-tabs__tab"
                  data-test="tab-2"
                  id="react-tabs-2"
                  role="tab"
                  tabIndex={null}
                >
                  Title 2
                </li>
              </Tab>
            </ul>
          </TabList>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-1"
            key=".1"
            selected={true}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-0"
          >
            <div
              aria-labelledby="react-tabs-0"
              className="react-tabs__tab-panel react-tabs__tab-panel--selected"
              id="react-tabs-1"
              role="tabpanel"
            >
              <h2>
                Any content 1
              </h2>
            </div>
          </TabPanel>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-3"
            key=".2"
            selected={false}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-2"
          >
            <div
              aria-labelledby="react-tabs-2"
              className="react-tabs__tab-panel"
              id="react-tabs-3"
              role="tabpanel"
            />
          </TabPanel>
        </div>
      </UncontrolledTabs>
    </Tabs>
  </ListTabs>
</App>
`;
